### PR TITLE
Feature/48272 jump to page

### PR DESCRIPTION
**Pull request by prasads-emgage**

### CODE REVIEW

File: example/src/App.tsx [hunk 1]
```diff
@@ -1343,7 +1343,7 @@ class App extends React.Component<{}, State> {
         onClick: (value: any) => {
           console.log('Version:', value);
         },
-      },
+      },104
```
Comment (ankitsaraogi) [inline example/src/App.tsx:1346]: what is this 104?

File: example/src/App.tsx [hunk 2]
```diff
@@ -1589,15 +1589,23 @@ class App extends React.Component<{}, State> {
...
           Pagination:
-          <Pagination {...PaginationDefaultProps} pageSizeList={[3, 6, 9]} showSizeChanger={true} onChange={(page: number) => this.setState({ paginationCurrent: page })} current={this.state.paginationCurrent} total={25} />
+          <Pagination {...PaginationDefaultProps} pageSizeList={[5, 10, 20, 50, 100]} showSizeChanger={true} onChange={(page: number) => this.setState({ paginationCurrent: page })} current={this.state.paginationCurrent} total={104}
+            showPrevNextJumpers={true}
+            showQuickJumper={true}
+            hideOnSinglePage={false}
+            showTitle={true}
+            simplePagination={false}
+            showJumpToPage={true}
+            // simple={false}
```
Comment (ankitsaraogi) [inline example/src/App.tsx:1599]: commented code should be removed

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -80,16 +86,19 @@ export const DefaultProps = {
   pageSizeOptions: [],
   showTotal: noop,
   pageSizeList: [],
-  lazyLoading: false
+  simple: false,
+  lazyLoading: false,
+  fixedSize: true,
+  simplePagination: false,
+  showJumpToPage: true
```
Comment (ankitsaraogi) [inline src/components/Pagination/Pagination.tsx:93]: lets keep showJumpToPage default to false

File: src/components/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -385,307 +415,625 @@ class Pagination extends React.PureComponent<IProps, IState> {
...
+          page={numPages}
+          className={theme['rc-pagination-disabled']}
+          showTitle={showTitle === true}
+          itemRender={itemRender}
+          componentStyle={{ transition: 'none' }}
+        />
+      );
+    }
+
+    const dataOrAriaAttributeProps = Object.keys(props).reduce(
```
Comment (ankitsaraogi) [inline src/components/Pagination/Pagination.tsx:993]: can we move this to a separate function and reuse wherever needed

File: src/components/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -385,307 +415,625 @@ class Pagination extends React.PureComponent<IProps, IState> {
...
+    const {
+      theme,
+      showLessItems,
+    } = this.props;
+
+    const { current = 1, pageSize } = this.state;
+
+    const numPages = calculatePage(undefined, this.state, this.props);
+
+    const maxPageLimit = showLessItems ? 7 : 9;
```
Comment (ankitsaraogi) [inline src/components/Pagination/Pagination.tsx:913]: how did we come to know numbers like 7 & 9

**Review by ankitsaraogi — CHANGES_REQUESTED** [review]

**Comment by emgageadmin** [comment]
https://atera.tpondemand.com/entity/48272

File: example/src/App.tsx (hunk 1 above)
Comment (prasads-emgage) [reply example/src/App.tsx:1346]: Will be removed..

File: example/src/App.tsx (hunk 2 above)
Comment (prasads-emgage) [reply example/src/App.tsx:1599]: Will be removed..

File: src/components/Pagination/Pagination.tsx (hunk 5 above)
Comment (prasads-emgage) [reply src/components/Pagination/Pagination.tsx:913]: We can make the maximum allowed Pager Buttons to 7. I think there is no need of 9 Buttons. The 9 button Pagination is by default enabled. Currently we disable it by setting the `'showLessItems' props to 'true'`

Screenshot 2021-04-04 at 4 51 27 PM

[https://material-ui.com/components/pagination/](https://material-ui.com/components/pagination/)


`[PREV]` ` [1]`  `[2]`  `[3]`  `[4]`  `[5]` ` [...]`  `[10]`  `[NEXT]`   |---> Selected Page 1, Total Page 10
`[PREV]`  `[1]`  `[...]`  `[5]`  `[6]`  `[7]`  `[...]`  `[10]`  `[NEXT]`  |---> Selected Page 6, Total Page 10
`[PREV]`  `[1]`  `[...]`  `[6]`  `[7]`  `[8]`  `[9]`  `[10]`  `[NEXT]`   |---> Selected Page 10, Total Page 10
`[PREV]`  `[1]`  `[2]`  `[3]`  `[4]`  `[5]`  `[6]`  `[7]`  `[NEXT]`      |---> Selected Page 4, Total Page 7
`[PREV]`  `[1]`  `[2]`  `[3]`  `[4]`  `[5]`  `[6]`  `[NEXT]`                |---> Selected Page 5, Total Page 6
`[PREV]`  `[1]`  `[2]`  `[3]`  `[4]`  `[5]`  `[NEXT]`                           |---> Selected Page 2, Total Page 5
`[PREV]`  `[1]`  `[2]`  `[3]`  `[4]`  `[NEXT]`                                     |---> Selected Page 2, Total Page 4
`[PREV]`  `[1]`  `[2]`  `[3]`  `[NEXT]`                                                |---> Selected Page 2, Total Page 3

**Review by ankitsaraogi — APPROVED** [review]

**Pull request merged**